File: Hash-Based Searching/Hash Table Lookup/HashTableLookup.py
TABLE_SIZE = 101
EMPTY = -1


class HashTableLookup:
    def __init__(self):
        self.table = [EMPTY] * TABLE_SIZE

    def hash_function(self, key):
        return key % TABLE_SIZE

    def insert(self, key):
        index = self.hash_function(key)

        if self.table[index] != EMPTY:
            return False

        self.table[index] = key
        return True

    def lookup(self, key):
        return (
            self.table[
                self.hash_function(key)
            ]
            == key
        )


if __name__ == "__main__":
    ht = HashTableLookup()

    ht.insert(15)
    ht.insert(42)
    ht.insert(99)

    print(
        "Found"
        if ht.lookup(42)
        else "Not Found"
    )

File: Hash-Based Searching/Hash Table Lookup/HashTableLookup.java
public class HashTableLookup {

    private static final int TABLE_SIZE = 101;
    private static final int EMPTY = -1;

    private final int[] table;

    public HashTableLookup() {
        table = new int[TABLE_SIZE];

        for (int i = 0; i < TABLE_SIZE; i++) {
            table[i] = EMPTY;
        }
    }

    private int hashFunction(int key) {
        return key % TABLE_SIZE;
    }

    public boolean insert(int key) {
        int index = hashFunction(key);

        if (table[index] != EMPTY)
            return false;

        table[index] = key;
        return true;
    }

    public boolean lookup(int key) {
        return table[hashFunction(key)] == key;
    }

    public static void main(String[] args) {
        HashTableLookup ht =
            new HashTableLookup();

        ht.insert(15);
        ht.insert(42);
        ht.insert(99);

        System.out.println(
            ht.lookup(42)
                ? "Found"
                : "Not Found"
        );
    }
}
